Add a React Native example app to the demo

jpmunz opened the pull request:
# Changes

Hello! I thought it would be valuable to include a React Native example app in this demo to show how mobile apps could be instrumented using the existing JS OTEL packages. I also thought this could serve as a bit of a TODO list showing some of the rough edges / missing functionality that current React Native developers would face when trying to instrument and hopefully keep updating this as those gaps are filled

A note on the size of the diff: unfortunately there's a ton of boilerplate files that are required for just spinning up a blank new app that make this harder to review. I have these on a separate commit but don't have permission to push that up as a branch, if someone could do that this PR could then be pointed to it and only show the files that are relevant to the demo similar to what you'd see on https://github.com/open-telemetry/opentelemetry-demo/pull/1781/commits/cde1345e47f06a4371394a7b3962a79a98449afd. Otherwise using the file filters to hide things like the .png, .xml, etc. will help narrow down to the relevant changes

Some notes on the overall approach here curious to hear thoughts or disagreement:
*  There's copy and pasting from `src/frontend` since that is also using React. However there were enough tweaks needed to make it work with React Native that it didn't seem straightforward to try and have these two apps share common files and also I think useful for demo purposes to have each be self-contained
* ~I haven't yet but I'm tempted to commit `protos/demo.ts` and `pb/demo.proto` just for this component, it's a bit weird since they are generated files but since this app isn't run inside Docker users will have to generate those files themselves which adds a bit of friction for running the demo~ committed now
* I only implemented a subset of the functionality demonstrated in the web app with this PR in the interest of hopefully getting something initial in place and tackling the rest later
* I haven't done the docs changes yet, would that be adding a new page under https://opentelemetry.io/docs/demo/services/ ? I can tackle that next if this addition overall seems reasonable

Using the demo:
![embrace](https://github.com/user-attachments/assets/1348bc75-0d53-452b-ad8a-a3ef6c62ff37)

## Merge Requirements

For new features contributions, please make sure you have completed the following
essential items:

* [x] `CHANGELOG.md` updated to document new feature additions
* [ ] Appropriate documentation updates in the [docs][]
* [ ] Appropriate Helm chart updates in the [helm-charts][]

<!--
A Pull Request that modifies instrumentation code will likely require an
update in docs. Please make sure to update the opentelemetry.io repo with any
docs changes.

A Pull Request that modifies docker-compose.yaml, otelcol-config.yaml, or
Grafana dashboards will likely require an update to the Demo Helm chart.
Other changes affecting how a service is deployed will also likely require an
update to the Demo Helm chart.
-->

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/
[helm-charts]: https://github.com/open-telemetry/opentelemetry-helm-charts

jpmunz commented on src/reactnativeapp/app/_layout.tsx:
```diff
@@ -0,0 +1,59 @@
+        <QueryClientProvider client={queryClient}>
+          <CartProvider>
+            {/*
+              TODO Once https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2359 is available it can
```
Mentioning https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2359 here so that the PRs get linked

julianocosta89 commented:
@jpmunz this looks great!

We had some similar work presented at a previous SIG meeting some time ago by @breedx-splk 's team.
Were you involved on that?

Are you able to join a OTel Demo SIG meeting to present this to everyone?

Also, adding a README on how to run the service would help a lot.

Regarding the doc page, yes, that would be under demo/services, but it may take a while to review this PR as it is a 25K lines of code addition.

breedx-splk commented:
> We had some similar work presented at a previous SIG meeting some time ago by @breedx-splk 's team. Were you involved on that?

I don't believe that they were...but this is super interesting. React native on mobile is asked about quite a lot -- so having this available in/through the android app would also be amazing and helpful. It would allow us to demonstrate the client-side RUM interop (currently nonexistent!) and compare/contrast features.

Logged this issue to track: https://github.com/open-telemetry/opentelemetry-android/issues/696

puckpuck commented:
> I haven't yet but I'm tempted to commit protos/demo.ts and pb/demo.proto just for this component, it's a bit weird since they are generated files but since this app isn't run inside Docker users will have to generate those files themselves which adds a bit of friction for running the demo

You can move forward with committing the generated proto files. We recently launched an effort to do the same for all other services in the Demo.

jpmunz commented:
> We had some similar work presented at a previous SIG meeting some time ago by @breedx-splk 's team. Were you involved on that?
> Are you able to join a OTel Demo SIG meeting to present this to everyone?

@julianocosta89 I was not involved in that previous meeting but I'm happy to join the next one (Nov 27th?) and present there!

> 
> Also, adding a README on how to run the service would help a lot.

I had included https://github.com/open-telemetry/opentelemetry-demo/blob/3404e12a242936bab02ad3a4d5e093fdb59839d1/src/reactnativeapp/README.md, did that include what you had in mind or was there something else I could add?

jpmunz commented:
> I don't believe that they were...but this is super interesting. React native on mobile is asked about quite a lot -- so having this available in/through the android app would also be amazing and helpful. It would allow us to demonstrate the client-side RUM interop (currently nonexistent!) and compare/contrast features.
> 
> Logged this issue to track: [open-telemetry/opentelemetry-android#696](https://github.com/open-telemetry/opentelemetry-android/issues/696)

@breedx-splk that's great would be cool if the UI here could be re-used to demo the Android SDK. Another possible avenue that I'll put out there could be using RN Native Modules to wrap the Android (and potentially Swift) OTEL SDKs in a thin JS package. That might be less flexible but could be appealing to devs who didn't want to deal with the setup on the native side and just wanted to add a JS package that handled it for them

julianocosta89 commented:
> > We had some similar work presented at a previous SIG meeting some time ago by @breedx-splk 's team. Were you involved on that?
> > Are you able to join a OTel Demo SIG meeting to present this to everyone?
> 
> @julianocosta89 I was not involved in that previous meeting but I'm happy to join the next one (Nov 27th?) and present there!

Yes! Nov 27th, but I'll try to put some time into reviewing the PR till there.

> > Also, adding a README on how to run the service would help a lot.
> 
> I had included https://github.com/open-telemetry/opentelemetry-demo/blob/3404e12a242936bab02ad3a4d5e093fdb59839d1/src/reactnativeapp/README.md, did that include what you had in mind or was there something else I could add?

This is perfect, I just missed it 😅

breedx-splk commented:
> using RN Native Modules to wrap the Android (and potentially Swift) OTEL SDKs in a thin JS package

Yeah I think we will need both native with RN inside and potentially this long term. There will always be devs that just write pure native Android, and we need to make sure we support them first.

julianocosta89 commented:
@jpmunz could you add some instructions on how to get this PR up and running to someone that has 0 knowledge on reactive native and mobile development?

Following the README didn't work for me, it seems that I'm missing 1 million tools to be able to run it.
Can I run it without the Android Emulator?

Also, the ios version, I'm getting this error:

```
> reactnativeapp@1.0.0 ios
> expo run:ios

env: load .env
env: export EXPO_PUBLIC_FRONTEND_PROXY_PORT
(node:8570) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
  CocoaPods CLI not found in your PATH, installing it now.
› Attempting to install CocoaPods CLI with Gem
Successfully installed cocoapods-1.16.2
1 gem installed
› Successfully installed CocoaPods CLI with Gem
✔ Installed CocoaPods CLI.
⚠️  Something went wrong running `pod install` in the `ios` directory.
Command `pod install` failed.
└─ Cause: spawn pod ENOENT
```

Not sure if that is something related to permissions. As I said, I have 0 knowledge on mobile development 😢

julianocosta89 commented:
@jpmunz I think I'll have to wait for your presentation on Wednesday 😢 

```
cp: invalid option -- 'X'
Try 'cp --help' for more information.
Command PhaseScriptExecution failed with a nonzero exit code

warning: Run script build phase '[CP-User] Generate app.config for prebuilt Constants.manifest' will be run during every build because it does not specify any outputs. To address this issue, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'EXConstants' from project 'Pods')
note: Run script build phase '[Expo] Configure project' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'reactnativeapp' from project 'reactnativeapp')
note: Run script build phase 'Bundle React Native code and images' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'reactnativeapp' from project 'reactnativeapp')
/Users/juliano.costa/workspace/opentelemetry-demo/src/reactnativeapp/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 12.0 to 18.1.99. (in target 'RNDeviceInfo-RNDeviceInfoPrivacyInfo' from project 'Pods')
/Users/juliano.costa/workspace/opentelemetry-demo/src/reactnativeapp/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 12.0 to 18.1.99. (in target 'RNCAsyncStorage-RNCAsyncStorage_resources' from project 'Pods')
warning: Run script build phase '[CP-User] [Hermes] Replace Hermes for the right configuration, if needed' will be run during every build because it does not specify any outputs. To address this issue, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'hermes-engine' from project 'Pods')


--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:iOS Simulator, id:D6417E92-19CA-498C-A079-CB35B4A56555, OS:18.1, name:iPhone SE (3rd generation) }
{ platform:iOS Simulator, id:D6417E92-19CA-498C-A079-CB35B4A56555, OS:18.1, name:iPhone SE (3rd generation) }
** BUILD FAILED **


The following build commands failed:
	PhaseScriptExecution [CP-User]\ Generate\ Specs /Users/juliano.costa/Library/Developer/Xcode/DerivedData/reactnativeapp-hfjcezijrrdpihfcaelpfppoavlp/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/React-Codegen.build/Script-46EB2E00019480.sh (in target 'React-Codegen' from project 'Pods')
	Building workspace reactnativeapp with scheme reactnativeapp and configuration Debug
(2 failures)
```

Now the dependencies are installed and theoretically all tools as well.

Not sure what I am missing.

julianocosta89 commented:
Ok, it seems that I was able to run it.
Now I'm getting a different error:
![Screenshot 2024-11-24 at 19 07 22](https://github.com/user-attachments/assets/2883dae2-9223-44f9-968b-086cd01d22cd)

```
No bundle URL present.

Make sure you're running a packager server or have included a .jsbundle file in your application bundle.

RCTFatal
__28-[RCTCxxBridge handleError:]_block_invoke
_dispatch_call_block_and_release
_dispatch_client_callout
_dispatch_main_queue_drain
_dispatch_main_queue_callback_4CF
__CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__
__CFRunLoopRun
CFRunLoopRunSpecific
GSEventRunModal
-[UIApplication _run]
UIApplicationMain
__debug_main_executable_dylib_entry_point
start_sim
0x0
0x0

```

pichlermarc commented on src/reactnativeapp/hooks/useTracer.ts:
```diff
@@ -0,0 +1,122 @@
+    [ATTR_DEVICE_ID]: getDeviceId(),
+  });
+
+  // TODO Not obvious that the WebTracerProvider can be used for React Native, might be useful to have a thin
+  //  ReactNativeTracerProvider on top of it (or BasicTracerProvider) that makes this clear. Could also add some
+  //  protection against browser specific functionality being added to WebTracerProvider that breaks functionality
+  //  for React Native.
+  //  Alternatively could offer a TracerProvider that exposed a JS interface on top of the OTEL Android and Swift SDKS,
+  //  giving developers the option of collecting telemetry at the native mobile layer
+  const provider = new WebTracerProvider({ resource });
```
Hmm, is there anything that would not work correctly when a `BasicTracerProvider` is used? 🤔

jpmunz replied:
so I seem to get less information using the `BasicTracerProvider`, e.g. for a request to the cart api I see this in Grafana when using it:
![image](https://github.com/user-attachments/assets/c498735a-bcab-4069-bf2c-c961aa5a0929)

but when using `WebTracerProvider` the trace includes the other services:
![image](https://github.com/user-attachments/assets/56d47f2c-27cd-4ad7-b423-f8519f4f39dc)

I think this might be due to the web one setting up a context manager (https://github.com/open-telemetry/opentelemetry-js/blob/main/packages/opentelemetry-sdk-trace-web/src/WebTracerProvider.ts#L60) whereas the basic one doesn't include any out of the box?

pichlermarc replied:
Yes that's likely it `ContextManager` and `Progagator` are missing with `BasicTracerProvider` and have to be registered manually. 

Probably fine to keep `WebTracerProvider` then as a first step as it's easier to use. 👍

jpmunz commented:
> Ok, it seems that I was able to run it. Now I'm getting a different error:

@julianocosta89 hmm so in that case it seems to have built the app correctly but can't load the JS bundle, in debug builds the bundle is grabbed from a local server instead of being packaged with the app. This local server should have been started automatically as part of `npm run android` but you can try spinning it up separately with `npm start`. Maybe also double-check the output from `npm run android` to see that the dev server was started correctly (maybe the port it tried using was occupied), it should look like:
![image](https://github.com/user-attachments/assets/59012074-cf71-40ba-a1e2-53be7fb2ecbb)

Hitting "Reload" on that error screen might also sometimes do the trick if the server is up but it wasn't able to connect with it right away

jpmunz commented on src/reactnativeapp/hooks/useTracer.ts:
```diff
@@ -0,0 +1,122 @@
+      // integration and make it less error-prone
+      getWebAutoInstrumentations({
+        "@opentelemetry/instrumentation-user-interaction": { enabled: false },
+        "@opentelemetry/instrumentation-document-load": { enabled: false },
```
ya good call I agree that's simpler, changed with https://github.com/open-telemetry/opentelemetry-demo/pull/1781/commits/2eace4abd0089d6fd7b043364fe0516b9058dde6, I mostly left the TODO as I think it's still relevant

jpmunz commented on src/reactnativeapp/hooks/useTracer.ts:
```diff
@@ -0,0 +1,122 @@
+  provider.addSpanProcessor(new SessionIdProcessor());
+
+  // Helpful for debugging
+  provider.addSpanProcessor(new SimpleSpanProcessor(new ConsoleSpanExporter()));
```
cool, switched here: https://github.com/open-telemetry/opentelemetry-demo/pull/1781/commits/b4dc87ebeac93353106ebdbc7d29eb402e49eca8

julianocosta89 commented:
@jpmunz running `npm start` before worked!
I wasn't able to run `npm run ios` though. I had to open Xcode and hit play, but maybe that's because I'm a newbie here 😅 

![image](https://github.com/user-attachments/assets/274a4806-b83b-4856-8e21-34d966f1da04)

jpmunz commented:
> @jpmunz running `npm start` before worked! I wasn't able to run `npm run ios` though. I had to open Xcode and hit play, but maybe that's because I'm a newbie here 😅

ah nice, glad it worked!

ya running it through Xcode can be helpful if there's a problem as it'll give more readable output than the CLI but then need to run the dev server separately (or do a release build). I also tend to run `rm -rf ~/Library/Developer/Xcode/DerivedData` every now and then as that seems to sometimes resolve unclear errors where Xcode has gotten confused by previous builds

julianocosta89 commented:
@jpmunz qq.

Shouldn't the `frontend-proxy ingress` be a child of `POST`?
![Screenshot 2024-11-27 at 19 08 19](https://github.com/user-attachments/assets/6b4a0380-2107-4bc4-9042-e4f90937edf1)

jpmunz commented:
@julianocosta89 @puckpuck I opened a draft PR for the docs changes here: https://github.com/open-telemetry/opentelemetry.io/pull/5704

jpmunz commented:
> @jpmunz qq.
> 
> Shouldn't the `frontend-proxy ingress` be a child of `POST`? ![Screenshot 2024-11-27 at 19 08 19](https://private-user-images.githubusercontent.com/15364991/390575184-6b4a0380-2107-4bc4-9042-e4f90937edf1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI4MDgzMzksIm5iZiI6MTczMjgwODAzOSwicGF0aCI6Ii8xNTM2NDk5MS8zOTA1NzUxODQtNmI0YTAzODAtMjEwNy00YmM0LTkwNDItZTRmOTA5MzdlZGYxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI4VDE1MzM1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE3YzYzMTczYjJhYWRmODRjNWZlMmM3NzZmYTg4YzQyOWE2MmVlOTNmNmM5MGU3NDdmODY1YThkNGIwOWI2NTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.goCQbC1HpdMjBsHDIKmg7vc_se-_LEZgbLwFfPRDR3E)

Hmm good call, ya when I do the equivalent on the browser it is a single trace:
![image](https://github.com/user-attachments/assets/012bf91e-8b7d-4550-aa31-4efc1f768fec)

Let me try and see what's happening there, everything under "reactnativeapp HTTP POST" looks correct but for some reason there's that standalone "reactnativeapp POST" that doesn't seem connected to anything

jpmunz commented:
> @jpmunz qq.
> 
> Shouldn't the `frontend-proxy ingress` be a child of `POST`?

@julianocosta89 looks to be due to a quirk with how React Native implements fetch, I think this should fix it: https://github.com/open-telemetry/opentelemetry-demo/pull/1781/commits/f1e8fc5f650a945f217df29cb8ab26eea837c911

CharlieTLe commented on src/reactnativeapp/pb/demo.proto:
```diff
@@ -0,0 +1,315 @@
+// Copyright 2020 Google LLC
```
We shouldn't check in this file since the original is from /pb/demo.proto and any changes to the original would cause this file to be updated, too.

jpmunz replied:
hey @CharlieTLe, this was discussed above with @puckpuck: https://github.com/open-telemetry/opentelemetry-demo/pull/1781#issuecomment-2484600350

Seems like we're moving towards having these proto files committed for the other services as well

CharlieTLe replied:
The `src/reactnativeapp/pb/demo.proto` file shouldn't be checked in since it's only the [Protobuf definition](https://protobuf.dev/overview/#syntax), and we have a copy of it in `pb/demo.proto`.

What should be checked in instead is the [generated proto code](https://protobuf.dev/programming-guides/proto3/#generated). So the file that should be checked in here is `src/reactnativeapp/protos/demo.ts` (which you've done). The `src/reactnativeapp/pb/demo.proto` should be added to the `.gitignore` file so that it doesn't accidentally get checked in.

Here's an [example](https://github.com/open-telemetry/opentelemetry-demo/pull/1794) of some generated proto code that should be checked in; it's for Python and Go, but the same idea still applies.

jpmunz replied:
ah sorry I had misunderstood, I think should be good now with: https://github.com/open-telemetry/opentelemetry-demo/pull/1781/commits/dbec3b7c36afa7d12cbb1b82f259ee4c3a9684d0

jpmunz commented:
@puckpuck @julianocosta89 as per discussion in the SIG I tried improving the react native instructions and also added a dockerfile for generating an apk here: https://github.com/open-telemetry/opentelemetry-demo/pull/1781/commits/9b6d92b8d3abdbdd7336dede558dad2c9e80069d

Some things to note:
* Realized that even with the apk generated in the container you'd still need some Android tooling available in order to run an emulator since the app only knows to hit localhost for the rest of the demo APIs. We talked about making that configurable as a follow-up but let me know if given that limitation it should be part of this initial PR
* For me the apk generation took ~15 minutes in the container vs. ~1 minute on host, I think that's due to a mix of gradle caching things, downloading extra tooling that dependencies require etc., there's likely a way to get this down more but I haven't dug into it further yet

julianocosta89 commented:
> @puckpuck @julianocosta89 as per discussion in the SIG I tried improving the react native instructions and also added a dockerfile for generating an apk here: [9b6d92b](https://github.com/open-telemetry/opentelemetry-demo/commit/9b6d92b8d3abdbdd7336dede558dad2c9e80069d)
> 
> Some things to note:
> 
> * Realized that even with the apk generated in the container you'd still need some Android tooling available in order to run an emulator since the app only knows to hit localhost for the rest of the demo APIs. We talked about making that configurable as a follow-up but let me know if given that limitation it should be part of this initial PR
> * For me the apk generation took ~15 minutes in the container vs. ~1 minute on host, I think that's due to a mix of gradle caching things, downloading extra tooling that dependencies require etc., there's likely a way to get this down more but I haven't dug into it further yet

Thanks for all the effort on this @jpmunz!
I've shared the app in the OTel Night in Berlin last week and everyone was super excited about it!

Regarding the docker step I have mixed feelings, 15 min seems like a lot 😬 

I'll go through the changes later today and try it out the new set of instructions on the README.

julianocosta89 commented on src/reactnativeapp/README.md:
````diff
@@ -0,0 +1,86 @@
+Or
+
+```bash
+make reactnative-run-ios
````
When running this I get the following error:

```
...
--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:iOS Simulator, id:FF9D57D3-43CB-42B6-8100-A49B4BEAB05D, OS:18.1, name:iPhone 16 Pro }
{ platform:iOS Simulator, id:FF9D57D3-43CB-42B6-8100-A49B4BEAB05D, OS:18.1, name:iPhone 16 Pro }
** BUILD FAILED **


The following build commands failed:
	PhaseScriptExecution [CP-User]\ Generate\ Specs /Users/juliano.costa/Library/Developer/Xcode/DerivedData/reactnativeapp-hfjcezijrrdpihfcaelpfppoavlp/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/React-Codegen.build/Script-46EB2E00019480.sh (in target 'React-Codegen' from project 'Pods')
	Building workspace reactnativeapp with scheme reactnativeapp and configuration Release
(2 failures)
Build logs written to /Users/juliano.costa/workspace/opentelemetry-demo/src/reactnativeapp/.expo/xcodebuild.log
make: *** [reactnative-run-ios] Error 1
```

The log file has 26700 lines. The `error` word is found 10113 times 🙃 .

I also get a lot of warnings like this:

```
/Users/juliano.costa/workspace/opentelemetry-demo/src/reactnativeapp/ios/Pods/Headers/Public/React-graphics/react/renderer/graphics/Transform.h:146:13: warning: parameter 'progress' not found in the function declaration [-Wdocumentation]
  146 |    * @param progress
      |             ^~~~~~~~
/Users/juliano.costa/workspace/opentelemetry-demo/src/reactnativeapp/ios/Pods/Headers/Public/React-graphics/react/renderer/graphics/Transform.h:146:13: note: did you mean 'animationProgress'?
  146 |    * @param progress
      |             ^~~~~~~~
      |             animationProgress
```

The `npm run ios` command also fails.

julianocosta89 replied:
@jpmunz I forgot to mention that I'm still able to run the project as I did before.

```
cd src/reactnativeapp
npm install
npm start
```

Then I go to XCode, build and run.

The main point of my comment was to say that we need to add those steps into the README, or change the current `make` command to run those steps.

jpmunz replied:
for `npm install` that I think should already be part of the make, the target has a dependency on the `node_modules` folder being there:
```
.PHONY: reactnative-run-ios
reactnative-run-ios: src/reactnativeapp/node_modules
  npm --prefix src/reactnativeapp run ios
```

which should then trigger the npm install:
```
src/reactnativeapp/node_modules:
  npm --prefix src/reactnativeapp install
```

`npm start` should only be required when doing a Debug build to serve up the JS bundle, to simplify things I had changed the commands to do release builds where the JS bundle is included as part of the app build. When you were able to build from XCode was that doing a Debug or Release configuration?

If it continues to cause problems I can switch things back to doing debug builds by default and add those steps + going to XCode as a troubleshooting fallback to the README (`npm run ios-debug` alone in theory should be enough as it both spins up the dev server and does the build in the same command)

julianocosta89 replied:
I'm just trying to follow the README.
When it doesn't work, I go to XCode to see if things still works.

In Xcode I can only run the app when I have the debug server running, so I guess I'm running debug mode.
But TBH I haven't configured anything, I just go there and hit play.

jpmunz replied:
ya would be the debug configuration by default in XCode, would have to run through the steps in those screenshots I posted above to swap it to the Release configuration

jpmunz replied:
@julianocosta89 maybe best if we could do a screenshare to go over the issues you're hitting so we can narrow down the improvements we should make to the README / commands? Could maybe use the Demo SIG meeting slot this Wednesday if the agenda is open, I'm also happy to schedule something separately if you're available

julianocosta89 replied:
Sounds good, I'll ping you on Slack

julianocosta89 commented on src/reactnativeapp/README.md:
````diff
@@ -0,0 +1,86 @@
+tools to be installed on your host. From the project root run:
+
+```bash
+make reactnative-build-android
````
Also got an error here:

```
android.Dockerfile:9
--------------------
   7 |     # Choosing a tag where the Android build tools match what we have in android/build.gradle to avoid the
   8 |     # container having to download them
   9 | >>> FROM reactnativecommunity/react-native-android:v13.2.1 AS builder
  10 |     
  11 |     WORKDIR /reactnativesrc/
--------------------
ERROR: failed to solve: reactnativecommunity/react-native-android:v13.2.1: no match for platform in manifest: not found

View build details: docker-desktop://dashboard/build/desktop-linux/desktop-linux/j8czacicufv0xervxcm6cw3kh
make: *** [reactnative-build-android] Error 1
```

jpmunz replied:
huh that one's strange, it seems stuck at grabbing the base image which should exist:
https://hub.docker.com/layers/reactnativecommunity/react-native-android/v13.2.1/images/sha256-9b94b331a83dceb53a3e6bb3b4afb4d29bd372ac78e18d33241199e6d2dad5d6?context=explore

I wonder if it's a host architecture difference, taking a look at some of the workarounds from https://stackoverflow.com/questions/76355896/docker-build-platform-linux-amd64-fails-error-failed-to-solve-no-match-for, you could try:
```
docker build -f src/reactnativeapp/android.Dockerfile --platform=linux/amd64 --output=. src/reactnativeapp
```

I wonder if that `platform` flag is needed for compatibility

julianocosta89 replied:
yes, platform flag solved the issue

julianocosta89 replied:
It took 16 minutes to execute on my M1 Max with 64GB of memory. I believe it will take longer in my Ubuntu with i5 7th gen 😅

julianocosta89 replied:
Also, not quite sure how that will help in the current state of the app, as we can't configure the frontend and Collector endpoints.

julianocosta89 replied:
Good point!
Maybe we can keep it, and then just point users to this page: https://developer.android.com/studio/run/emulator-install-add-files

So, for now, they would use the dockerfile to build the APK and Android Studio to run it in an emulated device locally.

Later we change that to something like "install in your personal phone".

julianocosta89 replied:
I'm okay in keeping it as is

jpmunz replied:
sounds good, I added that link to the README and that extra flag for the Docker command here: https://github.com/open-telemetry/opentelemetry-demo/pull/1781/commits/42af50d93b6fc6a3d6152f52c1e32919fe788ae1

jpmunz commented:
@julianocosta89 made some updates that we had discussed here: https://github.com/open-telemetry/opentelemetry-demo/pull/1781/commits/349e7c65e5e48ed17930ec95a0a997afd4936471

removed some of the make targets to simplify things, switched back to using debug builds and then added some troubleshooting steps to the readme

julianocosta89 commented on src/reactnativeapp/scripts/reset-project.js:
```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env node
```
Do we need this script?

jpmunz replied:
removed

julianocosta89 commented:
Hey @jpmunz, I went through all files and added a couple of nit.

Regarding the README I'm starting to wonder if we can't just go with the XCode steps.
I've tried all suggested steps, and all of them failed.

Just the last suggestion (open XCode and hit play) worked fine.
I'm worried about having multiple users trying to run the app and getting a failure, and opening issues in the demo repo.

jpmunz commented:
Thanks for the reviews @julianocosta89! Added these comments in response, I'll respond to some of the individual comments inline as well:
* adding line breaks -> https://github.com/open-telemetry/opentelemetry-demo/pull/1781/commits/95348c7b48fd930a3fdd099b9452243431b59a6b
* updating package + display name -> https://github.com/open-telemetry/opentelemetry-demo/pull/1781/commits/6c61944f15046a6f2fd74de6a2701411b40b415f
* updating images -> https://github.com/open-telemetry/opentelemetry-demo/pull/1781/commits/62d2efc37a1ede0c7f1ccb6d14bed6f71b3ab6ea
* removing debug lines -> https://github.com/open-telemetry/opentelemetry-demo/pull/1781/commits/ac0ff0e3dbb6dd333123fc52e75d0a87515e1a34
* updating ios instructions -> https://github.com/open-telemetry/opentelemetry-demo/pull/1781/commits/711dcb088759944b2d67491ceefcf3848be1b1c6

jpmunz commented on src/reactnativeapp/android/app/build.gradle:
```diff
@@ -0,0 +1,172 @@
+    buildToolsVersion rootProject.ext.buildToolsVersion
+    compileSdk rootProject.ext.compileSdkVersion
+
+    namespace 'com.anonymous.reactnativeapp'
```
for these I went with `io.opentelemetry.reactnativeapp` to keep with a domain controlled by opentelemetry

jpmunz commented on src/reactnativeapp/android/app/src/main/ic_launcher-playstore.png:
updated

jpmunz commented on src/reactnativeapp/android/app/src/main/AndroidManifest.xml:
```diff
@@ -0,0 +1,33 @@
+        <action android:name="android.intent.action.VIEW"/>
+        <category android:name="android.intent.category.DEFAULT"/>
+        <category android:name="android.intent.category.BROWSABLE"/>
+        <data android:scheme="myapp"/>
```
this is particular attribute is different than the display name but I updated elsewhere and went with "Astronomy Shop App"

jpmunz commented on src/reactnativeapp/gateways/Api.gateway.ts:
```diff
@@ -0,0 +1,150 @@
+/**
+ * Copied with modification from src/frontend/gateways/Api.gateway.ts
+ *
+ * TODO The React Native example only implements a subset of the functionality defined here, missing in particular is:
+ *  - showing ads
+ *  - showing recommendations
+ *  - calculating shipping costs
+ *  - currency conversion
+ */
```
we had discussed potentially creating a meta-issue to track all the react native related follow-up work, would that make sense to start now and include this as part of it or should I file just a standalone one?

jpmunz commented on src/reactnativeapp/hooks/useTracer.ts:
```diff
@@ -0,0 +1,128 @@
+      new SessionIdProcessor(),
+
+      // Helpful for debugging
+      new SimpleSpanProcessor(new ConsoleSpanExporter()),
```
makes sense, updated

jpmunz commented on src/reactnativeapp/ios/reactnativeapp/Images.xcassets/SplashScreenBackground.imageset/image.png:
I believe it's put behind the splash image when rendering the splash screen, in this case it's just putting the splash screen image on a white background but could be used to have something more complicated behind it

jpmunz commented on src/reactnativeapp/ios/reactnativeapp/Info.plist:
```diff
@@ -0,0 +1,77 @@
+      <dict>
+        <key>CFBundleURLSchemes</key>
+        <array>
+          <string>myapp</string>
```
similar here, the URL scheme is a bit different but I updated the display name separately

jpmunz commented on src/reactnativeapp/README.md:
````diff
@@ -0,0 +1,124 @@
+Or
+
+```bash
+npm run ios
````
for this one the error seems a bit more specific than the previous ones:
"iOS 18.2 is not installed. To use with Xcode, first download and install the platform"

this was a recent update for XCode that you may need to download, you could check which simulators you have installed by opening the IDE and clicking on the phone in the top middle bar then -> "Manage Run Destinations" -> "Simulators" and install anything that is shown as missing. You could also delete any extras there which may help

jpmunz commented:
> Hey @jpmunz, I went through all files and added a couple of nit.
> 
> Regarding the README I'm starting to wonder if we can't just go with the XCode steps. I've tried all suggested steps, and all of them failed.
> 
> Just the last suggestion (open XCode and hit play) worked fine. I'm worried about having multiple users trying to run the app and getting a failure, and opening issues in the demo repo.

hey @julianocosta89 yep that makes sense, I updated the readme here https://github.com/open-telemetry/opentelemetry-demo/pull/1781/commits/711dcb088759944b2d67491ceefcf3848be1b1c6 let me know if it addresses your concerns. I kept the `npm run ios` command as an option but I put the XCode steps as the first ones given

puckpuck commented:
I've spent the last several hours with this PR. Coming from no prior mobile dev environment on my local system, I was able to get the app working in Android and iOS with _minimal_ issues.

We can do quite a bit more to clarify the readme for this. I would also introduce a new `npm run start-ios,` which would make the XCode build version easier to understand. Finally, I want to rename the build Android make target to `build-react-native-android` instead so it is consistent with our other make targets.

I attempted to do the readme and DevX changes above but don't have permission to update the source repo. I can save the readme changes for a fast-follow PR, but I want the other DevX changes made to this PR.

puckpuck commented:
One more item is that I may want to rename the component/service and folder. We are currently embarking on an effort to have [consistent naming](https://github.com/open-telemetry/opentelemetry-demo/issues/1788) across all components in the demo. We want to enforce using a dash (`-`) in the component name where a space would be. Knowing this, is react native one or two words, and should we keep the `app` suffix or use `mobile` or nothing instead?

- `reactnative-app`
- `react-native-app`
- `reactnative-mobile`
- `react-native-mobile`
- `reactnative`
- `react-native`

@julianocosta89 do you have thoughts on this?

puckpuck commented on src/reactnativeapp/package.json:
```diff
@@ -0,0 +1,76 @@
+  "main": "expo-router/entry",
+  "version": "1.0.0",
+  "scripts": {
+    "start": "ENVFILE=.env expo start",
```
Can we add a `start-ios` target here as well to make it easy to start the ios specific JS bundle when used with XCode?

```suggestion
    "start": "ENVFILE=.env expo start",
    "start-ios": "ENVFILE=.env expo start --ios",
```

jpmunz replied:
this is actually a bit different of a way to run the app than the steps describe in the README, this launches a sandbox in the simulator called Expo Go which runs the JS bundle without building the app for iOS. It can be a quicker way to see things working but requires that the app not make use of any native code, I hadn't tried it for this app previously but looks like it fails due to some packages having native requirements

puckpuck replied:
The instructions in the readme say to start a dev service by running this command: `npm run start`. When I run that command, I'm presented with a menu that looks like this:

```› Using Expo Go
› Press s │ switch to development build

› Press a │ open Android
› Press i │ open iOS simulator
› Press w │ open web

› Press j │ open debugger
› Press r │ reload app
› Press m │ toggle menu
› Press o │ open project code in your editor
```
From here I hit `i` which ends up opening the iOS simulator with an Expo app with some JS errors. I switch over to XCode and hit the Play button; it builds the app and then starts it in the Simulator.

I assume these are the intended original steps. If so, I would like us to bypass forcing the user to press `i` to Open the iOS simulator.

jpmunz replied:
ah I see now, ya that can be ignored since hitting play through XCode will launch the simulator, I added a note about that here if that makes things clearer: https://github.com/open-telemetry/opentelemetry-demo/pull/1781/commits/9a190ce286e21c0926f054307eb41ab14b81d13a

puckpuck commented on Makefile:
```diff
@@ -191,3 +191,6 @@ else
 	@echo "Please provide a service name using `service=[service name]` or `SERVICE=[service name]`"
 endif
 
+.PHONY: reactnative-build-android
+reactnative-build-android:
```
rename this to `build-reactnative-android` to be consistent with our other make target names

```suggestion
.PHONY: build-reactnative-android
build-reactnative-android:
```

jpmunz replied:
when with `build-react-native-android` to match the other hyphenation: https://github.com/open-telemetry/opentelemetry-demo/pull/1781/commits/cd47bbc74767c0c8b3a25147a30197c80c7873c4

julianocosta89 commented:
> > Hey @jpmunz, I went through all files and added a couple of nit.
> > Regarding the README I'm starting to wonder if we can't just go with the XCode steps. I've tried all suggested steps, and all of them failed.
> > Just the last suggestion (open XCode and hit play) worked fine. I'm worried about having multiple users trying to run the app and getting a failure, and opening issues in the demo repo.
> 
> hey @julianocosta89 yep that makes sense, I updated the readme here [711dcb0](https://github.com/open-telemetry/opentelemetry-demo/commit/711dcb088759944b2d67491ceefcf3848be1b1c6) let me know if it addresses your concerns. I kept the `npm run ios` command as an option but I put the XCode steps as the first ones given

I've tested with a clean workspace and I actually need to first run the `npm run ios`, even though it fails, because that installs the Pods.
Without them, XCode just fail to build.

julianocosta89 commented:
> One more item is that I may want to rename the component/service and folder. We are currently embarking on an effort to have [consistent naming](https://github.com/open-telemetry/opentelemetry-demo/issues/1788) across all components in the demo. We want to enforce using a dash (`-`) in the component name where a space would be. Knowing this, is react native one or two words, and should we keep the `app` suffix or use `mobile` or nothing instead?
> 
> * `reactnative-app`
> * `react-native-app`
> * `reactnative-mobile`
> * `react-native-mobile`
> * `reactnative`
> * `react-native`
> 
> @julianocosta89 do you have thoughts on this?

I'd go vote for `react-native-app` or `astronomy-shop-app`.

jpmunz commented:
> > One more item is that I may want to rename the component/service and folder. We are currently embarking on an effort to have [consistent naming](https://github.com/open-telemetry/opentelemetry-demo/issues/1788) across all components in the demo. We want to enforce using a dash (`-`) in the component name where a space would be. Knowing this, is react native one or two words, and should we keep the `app` suffix or use `mobile` or nothing instead?
> > 
> > * `reactnative-app`
> > * `react-native-app`
> > * `reactnative-mobile`
> > * `react-native-mobile`
> > * `reactnative`
> > * `react-native`
> > 
> > @julianocosta89 do you have thoughts on this?
> 
> I'd go vote for `react-native-app` or `astronomy-shop-app`.

Updated to react-native-app: https://github.com/open-telemetry/opentelemetry-demo/pull/1781/commits/7b29f63b502d7e2aa703c244c4f0328ccd279560

Docs repo change: https://github.com/open-telemetry/opentelemetry.io/pull/5704/commits/61d52080a179f86ff24cfe8bb2c3946b7c03ca99

jpmunz commented:
> > > Hey @jpmunz, I went through all files and added a couple of nit.
> > > Regarding the README I'm starting to wonder if we can't just go with the XCode steps. I've tried all suggested steps, and all of them failed.
> > > Just the last suggestion (open XCode and hit play) worked fine. I'm worried about having multiple users trying to run the app and getting a failure, and opening issues in the demo repo.
> > 
> > 
> > hey @julianocosta89 yep that makes sense, I updated the readme here [711dcb0](https://github.com/open-telemetry/opentelemetry-demo/commit/711dcb088759944b2d67491ceefcf3848be1b1c6) let me know if it addresses your concerns. I kept the `npm run ios` command as an option but I put the XCode steps as the first ones given
> 
> I've tested with a clean workspace and I actually need to first run the `npm run ios`, even though it fails, because that installs the Pods. Without them, XCode just fail to build.

ah that's right, update instructions here: https://github.com/open-telemetry/opentelemetry-demo/pull/1781/commits/76e5cbe32ffde33751c09b21e9e5ceed5d3ead3f

jpmunz commented:
> Do we need the `ios/reactnativeapp.xcodeproj/project.pbxproj` and `ios/Podfile.lock` files part of this commit?
> 
> After resetting my environment and cloning this PR's branch, when I run this for iOS, those 2 files get updated. Should these 2 files be in `.gitignore`?

normally both these files are committed to version control, could you paste the diff you get after running these commands? I could take a look and try and see where the differences are coming from, there may be some tooling discrepancy which we may also have to lockdown

puckpuck commented:
I'm not sure if the rename broke it, but I can no longer build the app in XCode. I have tried to remove all Derived Data, and refresh the pod cache with a reinstall and still get the same error on Build
![Screenshot 2024-12-17 at 11 30 36 PM](https://github.com/user-attachments/assets/e82bfcf0-08a9-459b-922d-d284def69613)

julianocosta89 commented on src/react-native-app/gateways/Api.gateway.ts:
```diff
@@ -0,0 +1,150 @@
+/**
+ * Copied with modification from src/frontend/gateways/Api.gateway.ts
+ *
+ * TODO The React Native example only implements a subset of the functionality defined here, missing in particular is:
+ *  - showing ads
+ *  - showing recommendations
+ *  - calculating shipping costs
+ *  - currency conversion
+ */
```
Let's move this one to an issue.

jpmunz replied:
@julianocosta89 I had had a question around this above: https://github.com/open-telemetry/opentelemetry-demo/pull/1781#discussion_r1887669025, should it be a standalone issue or a meta issue tracking all React Native follow-ups?

puckpuck replied:
I created a meta-tracking [issue](https://github.com/open-telemetry/opentelemetry-demo/issues/1851) for this

julianocosta89 commented on src/react-native-app/ios/reactnativeapp/reactnativeapp.entitlements:
```diff
@@ -0,0 +1,5 @@
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+  <dict/>
+</plist>
```
Add line break

jpmunz replied:
updated

julianocosta89 reviewed (approved):
Add 3 comments, but to me it is ready to go!

julianocosta89 commented:
> hey @puckpuck normally both these files are committed to version control, could you paste the diff you get after running these commands?

I got the following:

```
diff --git a/src/react-native-app/ios/Podfile.lock b/src/react-native-app/ios/Podfile.lock
index 2188143..792a960 100644
--- a/src/react-native-app/ios/Podfile.lock
+++ b/src/react-native-app/ios/Podfile.lock
@@ -1602,8 +1602,8 @@ SPEC CHECKSUMS:
   RNReanimated: 35f9ac9c3ac42d0497ebd1cce5c39d7687a8493e
   RNScreens: b32a9ff15bea7fcdbe5dff6477bc503f792b1208
   SocketRocket: abac6f5de4d4d62d24e11868d7a2f427e0ef940d
-  Yoga: 2f71ecf38d934aecb366e686278102a51679c308
+  Yoga: ae3c32c514802d30f687a04a6a35b348506d411f
 
 PODFILE CHECKSUM: 43d62c9a2cdb594df87b33661537b45c69daf63f
 
-COCOAPODS: 1.15.2
+COCOAPODS: 1.16.2
```

jpmunz commented on src/react-native-app/README.md:
````diff
@@ -0,0 +1,136 @@
+Then install the pods for the project:
+
+```bash
+cd src/react-native-app/ios
````
updated

jpmunz commented:
> > hey @puckpuck normally both these files are committed to version control, could you paste the diff you get after running these commands?
> 
> I got the following:
> 
> ```
> diff --git a/src/react-native-app/ios/Podfile.lock b/src/react-native-app/ios/Podfile.lock
> index 2188143..792a960 100644
> --- a/src/react-native-app/ios/Podfile.lock
> +++ b/src/react-native-app/ios/Podfile.lock
> @@ -1602,8 +1602,8 @@ SPEC CHECKSUMS:
>    RNReanimated: 35f9ac9c3ac42d0497ebd1cce5c39d7687a8493e
>    RNScreens: b32a9ff15bea7fcdbe5dff6477bc503f792b1208
>    SocketRocket: abac6f5de4d4d62d24e11868d7a2f427e0ef940d
> -  Yoga: 2f71ecf38d934aecb366e686278102a51679c308
> +  Yoga: ae3c32c514802d30f687a04a6a35b348506d411f
>  
>  PODFILE CHECKSUM: 43d62c9a2cdb594df87b33661537b45c69daf63f
>  
> -COCOAPODS: 1.15.2
> +COCOAPODS: 1.16.2
> ```

ah ya we're not pinning the cocoapods version, I made an update here that hopefully locks that down: https://github.com/open-telemetry/opentelemetry-demo/pull/1781#discussion_r1887669025

I added a note to the README unfortunately it adds a few more steps to the setup on iOS so I worded it as something optional, let me know what you think

jpmunz commented:
> I'm not sure if the rename broke it, but I can no longer build the app in XCode. I have tried to remove all Derived Data, and refresh the pod cache with a reinstall and still get the same error on Build ![Screenshot 2024-12-17 at 11 30 36 PM](https://private-user-images.githubusercontent.com/1296118/396759533-e82bfcf0-08a9-459b-922d-d284def69613.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ1NTI0NzUsIm5iZiI6MTczNDU1MjE3NSwicGF0aCI6Ii8xMjk2MTE4LzM5Njc1OTUzMy1lODJiZmNmMC0wOGE5LTQ1OWItOTIyZC1kMjg0ZGVmNjk2MTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMThUMjAwMjU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTk1Yzc0MTM2ZmUzMGZjODlkMWY3ZTk3YmUzMzkzM2FkY2Y0MzZmZDdiYjk1ZDBkOWEyYjc3MzkyYzZjMzM0MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.fhk-y80GOoDDNOBkXRNPjoLNqlHCGAYaca4rrg6gYwM)

@puckpuck unfortunately I wasn't able to reproduce this, googling around a bit it seems like it could be an issue with the `.xcodeproj` file. Just to confirm when you open the project in XCode that you are opening `reactnativeapp.xcworkspace` as oppose to `reactnativeapp.xcodeproj`?

You had mentioned above seeing some local changes to `reactnativeapp.xcodeproj` were they still present? If you stash them do you get the same error?

puckpuck commented:
> @puckpuck unfortunately I wasn't able to reproduce this, googling around a bit it seems like it could be an issue with the `.xcodeproj` file. Just to confirm when you open the project in XCode that you are opening `reactnativeapp.xcworkspace` as oppose to `reactnativeapp.xcodeproj`?
> 
> You had mentioned above seeing some local changes to `reactnativeapp.xcodeproj` were they still present? If you stash them do you get the same error?

Yes, my issue was `xcodeproj` vs `xcworkspace`.

puckpuck commented:
I'm good with this. I want to make some changes to the Readme. We could have better structure and move some of the content into different sections, but I prefer we merge this first.

Thank you again, @jpmunz, for this fantastic PR! 🤩

julianocosta89 reviewed (approved):
@jpmunz would you be able to solve the merge conflicts?

This should be good to go!
Really appreciate your patience in guiding me through here, and all the back and forth 🙏🏽

jpmunz commented:
@julianocosta89 @puckpuck thank you both for your encouragement and taking the time to dive into and review this very large PR!

jpmunz commented:
@julianocosta89 merge conflicts should be resolved now

julianocosta89 commented:
@jpmunz it looks like there still conflict 🥲
I think it was my fault, I've just merged another PR.

If you could resolve it, I can get it merged right away

julianocosta89 commented:
Don't worry about the broken link. It is not related to your PR

jpmunz commented:
> @jpmunz it looks like there still conflict 🥲 I think it was my fault, I've just merged another PR.
> 
> If you could resolve it, I can get it merged right away

@julianocosta89 np, updated!

sRAVanI417 commented:
Hi, I am using a non expo based react-native project and I wanted a working demo for non expo react native application as it doesn't seem to work in that environment.

jpmunz commented:
> Hi, I am using a non expo based react-native project and I wanted a working demo for non expo react native application as it doesn't seem to work in that environment.

hey @sRAVanI417 I responded to your message in Slack but for tracking purposes I think best if you could open a new issue in this repo detailing the problems you are hitting
